sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala

```diff
@@ -194,4 +293,35 @@ class UserDefinedTypeSuite extends QueryTest with SharedSQLContext with ParquetT
     // call `collect` to make sure this query can pass analysis.
     pointsRDD.as[MyLabeledPoint].map(_.copy(label = 2.0)).collect()
   }
+
+  test("SPARK-19311: UDFs disregard UDT type hierarchy") {
+    UDTRegistration.register(classOf[IExampleBaseType].getName,
```

Comment:
oh. if you worry about that, actually we have `UDTRegistrationSuite` for test case of `UDTRegistration`. i am fine to either `SQLUserDefinedType` or `UDTRegistration`.